tests/test.validator.js

```diff
@@ -610,3 +610,41 @@ describe('Validator.detectTypeFromLayout()', function() {
       });
   });
 });
+
+
+describe('Validator.run()', function() {
+
+  it('should run extractMetaData()', () => {
+    var addonValidator = new Validator({_: ['foo'], metadata: true});
+    var fakeMetaData = {type: 1, somethingelse: 'whatever'};
+    addonValidator.toJSON = sinon.stub();
+    addonValidator.getAddonMetaData = () => {
+      return Promise.resolve(fakeMetaData);
+    };
+    addonValidator.checkFileExists = () => {
+      return Promise.resolve();
+    };
+    addonValidator.checkMinNodeVersion = () => {
+      return Promise.resolve();
+    };
+    class FakeXpi {
+      // stub Xpi class.
+    }
+    return addonValidator.run(FakeXpi)
+      .then(() => {
+        assert.ok(addonValidator.toJSON.called);
+        assert.deepEqual(
+          addonValidator.toJSON.firstCall.args[0].input, fakeMetaData);
+      });
+  });
+
+  it('should run scan()', () => {
+    var addonValidator = new Validator({_: ['foo'], metadata: false});
+    addonValidator.scan = sinon.stub();
```

Comment:
Again, newlines after `var` and before `return` make this a bit easier to scan visually.